Makefile: replace -G0 with -mno-gpopt

rofl0r opened the pull request:
like it's been done in most other pspdev repos.

ref https://github.com/pspdev/psptoolchain-allegrex/issues/24

rofl0r committed:
Makefile: replace -G0 with -mno-gpopt

like it's been done in most other pspdev repos.

ref https://github.com/pspdev/psptoolchain-allegrex/issues/24

diamant3 commented on Makefile:
```diff
@@ -10,7 +10,7 @@ vramalloc_OBJS = vramalloc.o
 
 INCDIR = 
-CFLAGS = -G0 -mno-explicit-relocs -O2 -Wall -Werror -g
+CFLAGS = -mno-gpopt -mno-explicit-relocs -O2 -Wall -Werror -g
```
could you remove the `-mno-explicit-relocs` and then do the same to CMakeLists.txt

rofl0r replied:
gonna do the cmake one in a separate pr since i used github edit frontend for this one, otherwise i'd have to switch to the git cli

diamant3 commented:
Thanks!, and yes this is one of the [packages](https://github.com/pspdev/psp-packages) needs an update regarding that issue

rofl0r committed:
Makefile: also remove -mno-gprelocs

while at it, remove the -Werror (compilers add new warnings, some of them false positives, all the time and we dont want the build to break because of them).

also make the psp-config invocations once, so they don't happen on each eval of the macros.
